### ISSUE TITLE
[OrderOfProperties]

### ISSUE THREAD

**Issue by AndrewPetrov** [realm/SwiftLint#2410]
### New Issue Checklist

- [ ] Updated SwiftLint to the latest version
- [ ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Please describe the rule idea, format
this issue's title as `Rule Request: [Rule Name]` and describe:

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.
2. Provide several examples of what _would_ and _wouldn't_ trigger violations.
3. Should the rule be configurable, if so what parameters should be configurable?
4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

I need rule to control properties groups order to keep file neat
All properties should be divided on several groups in order:

```
public let
public var

internal let
internal var

private let
private var

IBOutlet private var
```
The rule should triggering on mixing up this groups and on incorrect order
The rule should be configurable, I need to set desired `order`, or maybe order written above will bw enough
The rule should be opt-in

**Comment by kacper1703** [comment]
Any updates on that? Would love to have that too in my project.

**Comment by marcelofabri** [comment]
@kacper1703 feel free to open a PR! 💯

**Comment by piobyz** [comment]
This could be done on top of https://github.com/realm/SwiftLint/pull/2296 as suggested in https://github.com/realm/SwiftLint/issues/2699

**Comment by sarperdag** [comment]
Would really love to see this rule get implemented

**Comment by weakfl** [comment]
+1, this would be great
